hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1248,7 +1249,10 @@ private static Object rewritePrimaryTypeWithDiffSchemaType(Object oldValue, Sche
           return String.valueOf(oldValue);
         }
         if (oldSchema.getType() == Schema.Type.BYTES) {
-          return String.valueOf(oldValue);
+          if (oldValue instanceof ByteBuffer) {
```

Comment:
In my schema evolution pr I changed this to `StringUtils.fromUTF8Bytes(((ByteBuffer) oldValue).array());`. Will this do the same thing?